### ISSUE TITLE
Conflict with cline extension in VS Code Remote SSH: Multiple Users Unable to Use Extension Simultaneously

### ISSUE THREAD

**Issue by MineraleYT** [cline/cline#807]
### Which API Provider are you using?

Anthropic

### Which Model are you using?

Claude 3.5 Sonnet 20241022

### What happened?

Hello, I’m experiencing an issue with the cline extension when multiple users try to use it on the same remote machine via SSH using Visual Studio Code’s Remote Extensions.

**Environment:**
- Visual Studio Code Version: 1.95.3 Windows, 1.95.3 MacOS
- Cline Extension Version: 2.1.6
- Remote OS Version: Ubuntu 24.04.1 LTS
- Local OS Version (for both users): Windows 11 24H2, MacOS Sequoia 15.1.1
- VS Code Remote SSH Extension Version: v0.115.1

**Expected Behavior:**
User B should be able to use the cline extension independently without any errors.

**Actual Behavior:**
User B receives the following error message:
`Cannot register multiple views with same id claude-dev.SidebarProvider`
As a result, the cline extension does not function for User B.

**Questions:**
- Is there a way to configure the cline extension to support multiple users on the same remote machine?
- Are there any recommended workarounds to prevent this conflict?
- Does the extension support multi-user environments when using VS Code Remote SSH?

Thank you for your assistance.

### Steps to reproduce

1. User A connects to the remote machine via SSH in VS Code and installs the cline extension.
2. User A uses the cline extension without any problems.
3. User A disconnects from the remote session.
4. User B connects to the same remote machine via SSH in VS Code, with the cline extension already installed.
5. User B attempts to use the cline extension.

### Additional context

![cline](https://github.com/user-attachments/assets/e6fdce93-975b-4008-a426-7b28ae605a6d)

**Comment by saoudrizwan** [comment]
Thank you for reporting! I'll look into this and report back.

**Comment by johan456789** [comment]
I'd like to report a similar bug. It has nothing to do with SSH for me. I was editing two projects at the same time using two VS Code windows and running Cline in parallel in them. But one project's code appear in the other.

Simply put, cline doesn't support multitasking at all. It will mix up context and mess up. The issue is probably not related to VS Code Remote SSH.

Cline: 3.8.6
VS Code: 1.98.2
OS: macOS 15.3.2 (24D81)
Model I was using: gemini:gemini-2.5-pro-exp-03-25

@MineraleYT do you still have this issue? have you found a workaroud?

**Comment by MineraleYT** [comment]
> I'd like to report a similar bug. It has nothing to do with SSH for me. I was editing two projects at the same time using two VS Code windows and running Cline in parallel in them. But one project's code appear in the other.
> 
> Simply put, cline doesn't support multitasking at all. It will mix up context and mess up. The issue is probably not related to VS Code Remote SSH.
> 
> Cline: 3.8.6 VS Code: 1.98.2 OS: macOS 15.3.2 (24D81) Model I was using: gemini:gemini-2.5-pro-exp-03-25
> 
> [@MineraleYT](https://github.com/MineraleYT) do you still have this issue? have you found a workaroud?

Yes, I still have the issue and haven’t found any workaround yet.

**Comment by jgkme** [comment]
+1 facing same issue on remote workspace environment 
Only running 1 project at a time on visual studio code - the remote workspace also running locally on same pc

Cline: 3.15.2
VS Code: 1.100.1
OS: Win11 24H2
Model I was using: gemini 2.5 pro preview 03-25

**Comment by celestial-vault** [comment]
Thanks for reporting! This issue has been consolidated into the broader remote SSH workspace support feature request #4352. 

The specific multi-user/concurrent usage problem you've identified (the 'Cannot register multiple views with same id' error) affects both remote SSH and local multi-window scenarios, as noted in the comments. This will be addressed as part of the overall remote workspace improvements.
